### PR TITLE
[1068] Add support for displaying details on arbitrary element kinds

### PR DESCRIPTION
**Pull request by sbegaudeau**
- [1068] Add support for displaying details on arbitrary element kinds
- [test] Add additional tests to the collaborative form project

### CODE REVIEW

File: backend/sirius-components-collaborative-forms/src/main/java/org/eclipse/sirius/components/collaborative/forms/FormDescriptionAggregator.java [hunk 1]
```diff
@@ -33,34 +33,39 @@
  */
 public class FormDescriptionAggregator {
 
-    public Optional<FormDescription> aggregate(List<FormDescription> formDescriptions, Object object, IObjectService objectService) {
+    public Optional<FormDescription> aggregate(List<FormDescription> formDescriptions, List<Object> objects, IObjectService objectService) {
```
Comment (gcoutable) [inline backend/sirius-components-collaborative-forms/src/main/java/org/eclipse/sirius/components/collaborative/forms/FormDescriptionAggregator.java:36]: The class comment should be updated

File: backend/sirius-components-collaborative-forms/src/test/java/org/eclipse/sirius/components/collaborative/forms/FormDescriptionAggregatorTests.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        List<FormDescription> formDescriptions = new ArrayList<>();
+
+        FormDescription formDescription = this.createForm(object, 3, 2);
+        FormDescription formDescription2 = this.createForm(object, 3, 1);
+
+        formDescriptions.add(formDescription);
+        formDescriptions.add(formDescription2);
+
+        FormDescriptionAggregator aggregator = new FormDescriptionAggregator();
+        Optional<FormDescription> optional = aggregator.aggregate(formDescriptions, List.of(object), new IObjectService.NoOp());
```
Comment (gcoutable) [inline backend/sirius-components-collaborative-forms/src/test/java/org/eclipse/sirius/components/collaborative/forms/FormDescriptionAggregatorTests.java:50]: Is it relevant to test the behavior of the _FormDescriptionAggregator_ when there are many objects?

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/components/FormComponent.java [hunk 3]
```diff
@@ -45,10 +47,29 @@ public Element render() {
...
-            return new Element(PageComponent.class, pageComponentProps);
+        List<Element> children = candidates.stream().map(candidate -> {
+            VariableManager childVariableManager = variableManager.createChild();
+            childVariableManager.put(VariableManager.SELF, candidate);
+            return childVariableManager;
+        }).flatMap(childVariableManager -> {
+            return pageDescriptions.stream().map(pageDescription -> {
+                PageComponentProps pageComponentProps = new PageComponentProps(childVariableManager, pageDescription);
+                return new Element(PageComponent.class, pageComponentProps);
+            });
```
Comment (gcoutable) [inline backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/components/FormComponent.java:72]: There is something about the formatting, I cannot say exactly what. The formatting does not look clean.

File: frontend/src/properties/PropertiesWebSocketContainer.tsx [hunk 4]
```diff
@@ -132,8 +135,12 @@ export const PropertiesWebSocketContainer = ({
...
   }, [error, dispatch]);
 
-  let content = null;
-  if (!selection || propertiesWebSocketContainer === 'unsupportedSelection') {
+  let content: JSX.Element | null = null;
+  if (
+    propertiesWebSocketContainer === 'empty' ||
+    propertiesWebSocketContainer === 'unsupportedSelection' ||
+    propertiesWebSocketContainer === 'complete'
+  ) {
```
Comment (gcoutable) [inline frontend/src/properties/PropertiesWebSocketContainer.tsx:143]: What is the behavior when it's `idle` but there is no _form_?

File: frontend/src/properties/PropertiesWebSocketContainer.tsx [hunk 5]
```diff
@@ -22,7 +22,7 @@ import {
   subscribersUpdatedEventPayloadFragment,
   widgetSubscriptionsUpdatedEventPayloadFragment,
 } from 'form/FormEventFragments';
-import { GQLPropertiesEventSubscription } from 'form/FormEventFragments.types';
```
Comment (gcoutable) [inline frontend/src/properties/PropertiesWebSocketContainer.tsx:25]: Copyrigths

File: frontend/src/properties/PropertiesWebSocketContainerMachine.ts [hunk 6]
```diff
@@ -20,7 +20,7 @@ import {
   GQLWidgetSubscriptionsUpdatedEventPayload,
 } from 'form/FormEventFragments.types';
 import { v4 as uuid } from 'uuid';
-import { SelectionEntry } from 'workbench/Workbench.types';
```
Comment (gcoutable) [inline frontend/src/properties/PropertiesWebSocketContainerMachine.ts:23]: Copyrights

File: backend/sirius-components-collaborative-forms/src/test/java/org/eclipse/sirius/components/collaborative/forms/FormDescriptionAggregatorTests.java (hunk 2 above)
Comment (sbegaudeau) [reply backend/sirius-components-collaborative-forms/src/test/java/org/eclipse/sirius/components/collaborative/forms/FormDescriptionAggregatorTests.java:50]: We would only test that we iterate properly on a for loop somewhere. I wanted to have at least one test here for now.

File: backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/components/FormComponent.java (hunk 3 above)
Comment (sbegaudeau) [reply backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/components/FormComponent.java:72]: The formatting matches what we do most of the time (here, it's the flatMap with the sub stream inside which can make it look odd). I can remove an useless `return` and simplify the indentation.

**Review by gcoutable — APPROVED** [review]

**Pull request merged**